### PR TITLE
show inventory parser fix ax external user

### PR DESCRIPTION
**Pull request by kamyarz1991**

### CODE REVIEW

File: src/genie/libs/parser/iosxe/tests/test_show_platform.py [hunk 1]
```diff
@@ -2220,7 +2220,7 @@ class TestShowInventory(unittest.TestCase):
                 }
             },
             'F0': {
-                'other': {
+                'lc': {
```
[the remark this answers is not in the document]
Comment (kamyarz1991) [reply src/genie/libs/parser/iosxe/tests/test_show_platform.py]: Done!

File: src/genie/libs/parser/iosxe/tests/test_show_platform.py [hunk 2]
```diff
@@ -2316,111 +2316,123 @@ class TestShowInventory(unittest.TestCase):
...
+                    "ACS-4330-FANASSY":{
+                    "descr":"Cisco ISR4330 Fan Assembly",
+                    "name":"Fan Tray",
+                    "pid":"ACS-4330-FANASSY",
+                    "sn":"",
+                    "vid":""
+                    }
+                }
+            },
+            "P0":{
```
[the remark this answers is not in the document]
Comment (kamyarz1991) [reply src/genie/libs/parser/iosxe/tests/test_show_platform.py]: We do have code for that 
`

                # Power Supply Module 0
                # Power Supply Module 1
                p1_2 = re.compile(r'Power Supply Module')
                m1_2 = p1_2.match(name)
                if m1_2:
                    slot = name.replace('Power Supply Module ', 'P')
                    # Creat slot_dict
                    slot_dict = ret_dict.setdefault(
                        'slot', {}).setdefault(slot, {})
`

[the remark this answers is not in the document]
Comment (kamyarz1991) [reply src/genie/libs/parser/iosxe/tests/test_show_platform.py]: line 1863  of show_platform.py module

File: src/genie/libs/parser/iosxe/tests/test_show_platform.py [hunk 3]
```diff
@@ -2316,111 +2316,123 @@ class TestShowInventory(unittest.TestCase):
...
+                    "descr":"Cisco ISR4331 Built-In SM controller",
+                    "name":"module 1",
+                    "pid":"ISR4331/K9",
+                    "sn":"",
+                    "vid":""
+                    }
+                }
             },
+            "F0":{
+                "lc":{
```
[the remark this answers is not in the document]
Comment (kamyarz1991) [reply src/genie/libs/parser/iosxe/tests/test_show_platform.py]: Done!

File: src/genie/libs/parser/iosxe/tests/test_show_platform.py [hunk 4]
```diff
@@ -2924,6 +3056,190 @@ class TestShowInventory(unittest.TestCase):
...
+                        },
+                        "3 disk1":{
+                            "PVDM4-32":{
+                                "descr":"PVDM4-32 Voice DSP Module",
+                                "name":"PVDM subslot 0/4",
+                                "pid":"PVDM4-32",
+                                "sn":"FOC17176BKC",
+                                "vid":"V01"
+                            }
+                        }
```
[the remark this answers is not in the document]
Comment (kamyarz1991) [reply src/genie/libs/parser/iosxe/tests/test_show_platform.py]: Fixed

File: src/genie/libs/parser/iosxe/tests/test_show_platform.py [hunk 5]
```diff
@@ -2709,6 +2721,126 @@ class TestShowInventory(unittest.TestCase):
...
+                                    "sn":"FOC21032R9L",
+                                    "vid":"V04"
+                                }
+                            }
+                        }
+                    }
+                }
+            },
+            "F0":{
+                "lc":{
```
[the remark this answers is not in the document]
Comment (kamyarz1991) [reply src/genie/libs/parser/iosxe/tests/test_show_platform.py]: done

File: src/genie/libs/parser/iosxe/tests/test_show_platform.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tahigash) [reply src/genie/libs/parser/iosxe/tests/test_show_platform.py]: confirmed

File: src/genie/libs/parser/iosxe/tests/test_show_platform.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tahigash) [reply src/genie/libs/parser/iosxe/tests/test_show_platform.py]: got it. confirmed

File: src/genie/libs/parser/iosxe/tests/test_show_platform.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (tahigash) [reply src/genie/libs/parser/iosxe/tests/test_show_platform.py]: confirmed

File: src/genie/libs/parser/iosxe/tests/test_show_platform.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (tahigash) [reply src/genie/libs/parser/iosxe/tests/test_show_platform.py]: confirmed

**Comment by tahigash** [comment]
looks good now

**Pull request merged**